plugins/module_utils/config_utils.py

```diff
@@ -99,3 +104,192 @@ def changed(self) -> bool:
+        exhausts all possibilities.
+
+        Parameters:
+        - dictionary (dict): The dictionary to search through. This can be a multi-level nested dictionary.
+        - key (str): The key to search for in the dictionary.
+
+        Returns:
+        - The value associated with the found key, or None if the key is not found in the dictionary.
+
+        Example:
+        - Given the dictionary `{"interfaces": {"wan": {"if": 'interfaces/wan/if'}}}` and the key `'if'`,
+        the function will return `'interfaces/wan/if'`.
+        """
+
+        if isinstance(dictionary, dict):
+            for k, v in dictionary.items():
+                if k == key:
+                    return v
+                else:
+                    result = self._search_map(v, key)
+                    if result is not None:
+                        return result
+
+    def _get_xpath(self, module: str = None, setting: str = None, map_dict=None):
+        """
+        Retrieve the XPath for a given module and setting from the version-specific mapping.
+
+        This method takes a module and setting as input and looks up their corresponding XPath
+        in a version-specific mapping dictionary. It performs a direct lookup if the setting is
+        top-level within the module or initiates a recursive search if the setting is nested.
+
+        Parameters:
+        - module (str, optional): The name of the module to look up. Defaults to None.
+        - setting (str, optional): The name of the setting within the module to look up. Defaults to None.
+        - map_dict (dict, optional): A pre-fetched version map dictionary. If not provided,
+        it will use the version_map attribute based on the instance's version attribute. Defaults to None.
+
+        Returns:
+        - str or None: The XPath as a string if the module and setting are found within the map. If the
+        module or setting cannot be found, or the inputs are None, it returns None.
+
+        Example:
+        - For an instance with version 'OPNsense 23.1', if version_map['OPNsense 23.1'] contains {'system': {'hostname': 'system/hostname'}},
+        and we call _get_xpath(module='system', setting='hostname'), it will return 'system/hostname'.
+        """
+
+        map_dict = self.version_map.get(self.version)
+
+        if map_dict is None:
+            raise KeyError(f"{self.version} was not not found in version_map")
+
+        # Check if the provided module is in the map
+        if module in map_dict:
+            # If the setting is directly within the module, return it
+            if setting in map_dict[module]:
+                return map_dict[module][setting]
+            # If the setting is nested, search recursively
+            else:
+                return self._search_map(map_dict[module], setting)
+
+        return None
+
+    def _get_php_requirements(self, module: str = None, setting: str = None) -> list:
```

Comment:
```suggestion
    def _get_php_requirements(self, module: str = None, setting: str = None) -> Union[list | None]:

Reply:
I absolutely agree with you and thank you for pointing out this inconsistency. I have ensured consistent typing across the functions.